ouroboros-consensus/test-storage/Test/Ouroboros/Storage/TestBlock.hs

```diff
@@ -0,0 +1,384 @@
+import qualified Data.Map.Strict as Map
+import           Data.Maybe (maybeToList)
+import           Data.Word
+import           GHC.Generics (Generic)
+import           GHC.Stack (HasCallStack)
+import           Test.QuickCheck
+
+import           Control.Monad.Class.MonadThrow
+
+import           Cardano.Crypto.DSIGN
+import           Cardano.Prelude (NoUnexpectedThunks)
+
+import           Ouroboros.Network.Block
+import           Ouroboros.Network.MockChain.Chain (Point)
+import qualified Ouroboros.Network.MockChain.Chain as Chain
+
+import           Ouroboros.Consensus.Block
+import           Ouroboros.Consensus.Ledger.Abstract
+import           Ouroboros.Consensus.Ledger.Extended
+import           Ouroboros.Consensus.NodeId (NodeId (..))
+import           Ouroboros.Consensus.Protocol.Abstract
+import           Ouroboros.Consensus.Protocol.BFT
+import           Ouroboros.Consensus.Protocol.Signed
+import           Ouroboros.Consensus.Util.Condense
+import           Ouroboros.Consensus.Util.Orphans ()
+import qualified Ouroboros.Consensus.Util.SlotBounded as SB
+
+import           Ouroboros.Storage.FS.API (HasFS (..), withFile)
+import           Ouroboros.Storage.FS.API.Types
+
+import           Test.Util.Orphans.Arbitrary ()
+
+{-------------------------------------------------------------------------------
+  TestBlock
+-------------------------------------------------------------------------------}
+
+data TestBlock = TestBlock {
+      testHeader :: !TestHeader
+    , testBody   :: !TestBody
+    }
+  deriving stock    (Show, Eq, Generic)
+  deriving anyclass (NoUnexpectedThunks, Serialise)
+
+-- | Hash of a 'TestHeader'
+newtype TestHeaderHash = TestHeaderHash Int
+  deriving stock    (Eq, Ord, Show, Generic)
+  deriving newtype  (Condense, NoUnexpectedThunks, Hashable, Serialise)
+
+-- | Hash of a 'TestBody'
+newtype TestBodyHash = TestBodyHash Int
+  deriving stock    (Eq, Ord, Show, Generic)
+  deriving newtype  (Condense, NoUnexpectedThunks, Hashable, Serialise)
+
+data TestHeader = TestHeader {
+      thHash     :: HeaderHash TestHeader
+      -- ^ Not included in the calculation of the hash of the 'TestHeader',
+      -- i.e., in its own value, which would be pretty hard to do.
+      --
+      -- Note the absence of a bang: this field caches the 'TestHeader's hash.
+      -- To calculate it, the 'TestHeader' is passed to the hashing function,
+      -- even though the field is not read, making the field strict would
+      -- create an infinite loop.
+    , thPrevHash :: !(ChainHash TestHeader)
+    , thBodyHash :: !TestBodyHash
+    , thSlotNo   :: !SlotNo
+    , thBlockNo  :: !BlockNo
+    , thIsEBB    :: !Bool
+    }
+  deriving stock    (Eq, Show, Generic)
+  deriving anyclass (NoUnexpectedThunks, Serialise)
+
+data TestBody = TestBody {
+      tbForkNo  :: !Word
+      -- ^ If we don't have something that can vary per block, we're not
+      -- generating forks, except when skipping slots.
```

Comment:
Let's document that this is a _local_ number. Perhaps even giving a small example would help (with at least two split points in the tree, to make clear that this is indeed local info).